salt/utils/platform.py

```diff
@@ -91,16 +91,17 @@ def is_smartos_globalzone():
     if not is_smartos():
         return False
     else:
-        cmd = ['zonename']
         try:
-            zonename = subprocess.Popen(
-                cmd, shell=False,
-                stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            zonename_proc = subprocess.Popen(
+                ['zonename'], stdout=subprocess.PIPE, stderr=subprocess.STDOUT,
+            )
+            zonename_output = zonename_proc.communicate()[0].strip().decode("utf-8")
```

Comment:
pretty sure this should be [`__salt_system_encoding__`](https://docs.saltstack.com/en/latest/ref/states/writing.html#strings-and-unicode)

Reply:
The docs seems to indicate this is specifically for encoding when writing out str/unicode data to disks? Here we are ingesting data on a lower level (aka not wrapping salt.module.cmdmod.run() because it is not yet available, which takes care of these encoding bits for us), so we do the decode to ensure we feed unicode data to the rest of salt.

Reply:
The docs seems to indicate this is specifically for encoding when writing out str/unicode data to disks? Here we are ingesting data on a lower level (aka not wrapping salt.module.cmdmod.run() because it is not yet available, which takes care of these encoding bits for us), so we do the decode to ensure we feed unicode data to the rest of salt.

Reply:
I found a few locations were this is also use with decode, I will change it and retest.

Reply:
At a quick glance it looks like `__salt_system_encoding__` is undefiend for salt/utils/platform.py, maybe because it is so low level most of the salt objects like __salt__ and friends also don't exist yet?

Reply:
If the system isn't utf-8, then decoding as utf-8 *could* bork up the encoding. I mean, it *should* be utf-8 (and use UTC), but not all sysadmins have the luxury of dealing with sane systems ;)